components/skus/renderer/skus_page_controller.cc

```diff
@@ -0,0 +1,353 @@
+      v8::Promise::Resolver::New(isolate->GetCurrentContext());
+  if (resolver.IsEmpty()) {
+    return v8::Local<v8::Promise>();
+  }
+
+  auto promise_resolver(
+      v8::Global<v8::Promise::Resolver>(isolate, resolver.ToLocalChecked()));
+  auto context_old(
+      v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+
+  skus_service_->PrepareCredentialsPresentation(
+      domain, path,
+      base::BindOnce(&SkusPageController::OnPrepareCredentialsPresentation,
+                     base::Unretained(this), std::move(promise_resolver),
+                     isolate, std::move(context_old)));
+
+  return resolver.ToLocalChecked()->GetPromise();
+}
+
+void SkusPageController::OnPrepareCredentialsPresentation(
+    v8::Global<v8::Promise::Resolver> promise_resolver,
+    v8::Isolate* isolate,
+    v8::Global<v8::Context> context_old,
+    const std::string& response) {
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context = context_old.Get(isolate);
+  v8::Context::Scope context_scope(context);
+  v8::MicrotasksScope microtasks(isolate,
+                                 v8::MicrotasksScope::kDoNotRunMicrotasks);
+
+  v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
+  v8::Local<v8::String> result;
+  result = v8::String::NewFromUtf8(isolate, response.c_str()).ToLocalChecked();
+
+  ALLOW_UNUSED_LOCAL(resolver->Resolve(context, result));
+}
+
+// window.chrome.braveSkus.credential_summary
+v8::Local<v8::Promise> SkusPageController::CredentialSummary(
+    v8::Isolate* isolate,
+    std::string domain) {
+  if (!EnsureConnected())
+    return v8::Local<v8::Promise>();
+
+  v8::MaybeLocal<v8::Promise::Resolver> resolver =
+      v8::Promise::Resolver::New(isolate->GetCurrentContext());
+  if (resolver.IsEmpty()) {
+    return v8::Local<v8::Promise>();
+  }
+
+  auto promise_resolver(
+      v8::Global<v8::Promise::Resolver>(isolate, resolver.ToLocalChecked()));
+  auto context_old(
+      v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+
+  skus_service_->CredentialSummary(
+      domain,
+      base::BindOnce(&SkusPageController::OnCredentialSummary,
+                     base::Unretained(this), std::move(promise_resolver),
+                     isolate, std::move(context_old)));
+
+  return resolver.ToLocalChecked()->GetPromise();
+}
+
+void SkusPageController::OnCredentialSummary(
+    v8::Global<v8::Promise::Resolver> promise_resolver,
+    v8::Isolate* isolate,
+    v8::Global<v8::Context> context_old,
+    const std::string& response) {
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context = context_old.Get(isolate);
+  v8::Context::Scope context_scope(context);
+  v8::MicrotasksScope microtasks(isolate,
+                                 v8::MicrotasksScope::kDoNotRunMicrotasks);
+
+  v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
+
+  base::JSONReader::ValueWithError value_with_error =
```

Comment:
same as above re: DictionaryValue, but what are the other responses in these methods?